components/learn/modules/ROOT/pages/write-contracts.adoc

```diff
@@ -0,0 +1,158 @@
+
+With that out of the way, let's get started!
+
+[[setting-up-a-solidity-project]]
+== Setting up a Solidity Project
+
+The first step in any project is to create a `contracts` directory, which is where you will store your Solidity source files (`.sol`). This is equivalent to the `src` directory you may be familiar with from other languages.
+
+We can now write our first simple smart contract:
+
+```solidity
+pragma solidity ^0.5.0;
+
+contract SetAndGet {
+    uint256 private value;
+
+    // Emitted when the stored value changes
+    event ValueChanged(uint256 newValue);
+
+    // Stores a new value in the contract
+    function set(uint256 newValue) public {
+      value = newValue;
+
+      emit ValueChanged(newValue);
+    }
+
+    // Reads the last stored value
+    function get() public view returns (uint256) {
+      return value;
+    }
+}
+```
+
+We will store it in `contracts/StoreAndGet.sol`. Each `.sol` file should have the code for a single contract, and be named after it.
+
+[[compiling-solidity-source-code]]
+== Compiling Solidity Source Code
+
+The Ethereum Virtual Machine (EVM) cannot execute Solidity code directly: we first need to compile it into EVM bytecode.
+
+To do so, we'll use the https://docs.openzeppelin.com/sdk/2.6/[*OpenZeppelin command-line interface (CLI)*]:
+
+```bash
+$ npm install --save-dev @openzeppelin/cli
+```
+
+The CLI is quite powerful, and it is the entry point to many exciting features (such as upgradeable smart contracts and interactive transactions!). We will come back to it often.
+
+For now however, let's focus on compiling. This can be achieved by running a single command:
+
+```bash
+$ npx oz compile
+```
+
+NOTE: If you're unfamiliar with the `npx` command, check out our https://example.com[project setup guide].
+
+The `compile` command will automatically look for all contracts in the `contracts` directory, and compile them using the Solidity compiler version that best suits your project.
+
+You will notice a `build/contracts` directory was created: it holds the compiled artifacts (bytecode and metadata), which are `.json` files. It's a good idea to add this directory to your `.gitignore`.
+
+You can also configure your compilation, which includes choosing a compiler version and enabling optimizations, by passing arguments to the `compile` command:
+
+```bash
+$ npx oz compile --solc-version=0.5.12 --optimizer on
+```
+
+For detailed information on these options, refer to the https://docs.openzeppelin.com/sdk/2.6/api/cli#compile[`compile` API reference].
+
+[[multi-file-projects]]
+== Multi-File Projects
+
+There are multiple reasons that cause most projects to have multiple Solidity source files: from modularization via https://solidity.readthedocs.io/en/latest/contracts.html#inheritance[inheritance] to complex multi-contract systems.
```

Comment:
Let's tone this down. We can just state that _as our project grows, we will need to create more contracts, which may interact with each other_.

Reply:
'Tone down' as in make it simpler?

Reply:
Yes, _multi-file projects_ sounds intimidating, or even enterprisey. We are just talking about throwing in more files to our project.